src/d3a/models/strategy/external_strategies/pv.py

```diff
@@ -321,7 +338,14 @@ def _update_offer_aggregator(self, arguments):
                     continue
 
     def _offer_aggregator(self, arguments):
-        assert set(arguments.keys()) == {'price', 'energy', 'transaction_id', 'type'}
+        required_args = {'price', 'energy', 'type', 'transaction_id'}
+        allowed_args = required_args.union({'replace_existing'})
+
+        # Check that all required arguments have been provided
+        assert all(arg in arguments.keys() for arg in required_args)
+        # Check that every provided argument is allowed
+        assert all(arg in allowed_args for arg in arguments.keys())
```

Comment:
I see that this is used more than 5 times in the PR, might be a good candidate for a reusable function

Reply:
I agree: this should be refactored. However, I think that the refactoring should do more than just moving these things around: it would be good to have a validator class (or something similar) that can separate this logic from the external strategies classes. That's why I think it would be a candidate for a future task.